ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/marmot/results/RaoResultArchiveManager.java

```diff
@@ -0,0 +1,99 @@
+        String jsonFileNameTemplate = getIndividualRaoResultFilenameTemplate(properties);
+        String summaryFilename = getSummaryFilename(properties);
+        for (OffsetDateTime timestamp : interTemporalRaoResult.getTimestamps()) {
+            addRaoResultToZipArchive(timestamp, zipOutputStream, interTemporalRaoResult.getIndividualRaoResult(timestamp), cracs.getData(timestamp).orElseThrow(), properties, jsonFileNameTemplate);
+        }
+        addSummaryToZipArchive(zipOutputStream, interTemporalRaoResult, summaryFilename, jsonFileNameTemplate);
+        zipOutputStream.close();
+    }
+
+    private static void addRaoResultToZipArchive(OffsetDateTime timestamp, ZipOutputStream zipOutputStream, RaoResult raoResult, Crac crac, Properties properties, String jsonFileNameTemplate) throws IOException {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        raoResult.write("JSON", crac, properties, byteArrayOutputStream);
+        addEntryToZipArchive(timestamp.format(DateTimeFormatter.ofPattern(jsonFileNameTemplate)), zipOutputStream, byteArrayOutputStream);
+    }
+
+    private static void addSummaryToZipArchive(ZipOutputStream zipOutputStream, InterTemporalRaoResult interTemporalRaoResult, String summaryFilename, String jsonFileNameTemplate) throws IOException {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        try {
+            ObjectMapper objectMapper = JsonUtil.createObjectMapper();
+            SimpleModule module = new JsonInterTemporalRaoResultSerializerModule(jsonFileNameTemplate);
+            objectMapper.registerModule(module);
+            ObjectWriter writer = objectMapper.writerWithDefaultPrettyPrinter();
+            writer.writeValue(byteArrayOutputStream, interTemporalRaoResult);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        addEntryToZipArchive(summaryFilename, zipOutputStream, byteArrayOutputStream);
+    }
+
+    private static void addEntryToZipArchive(String entryName, ZipOutputStream zipOutputStream, ByteArrayOutputStream byteArrayOutputStream) throws IOException {
+        ZipEntry entry = new ZipEntry(entryName);
+        zipOutputStream.putNextEntry(entry);
+        addOutputStreamContentToZipAndClose(zipOutputStream, byteArrayOutputStream);
+        zipOutputStream.closeEntry();
+    }
+
+    private static void addOutputStreamContentToZipAndClose(ZipOutputStream zipOutputStream, ByteArrayOutputStream byteArrayOutputStream) throws IOException {
+        byte[] bytes = new byte[1024];
+        int length;
+        InputStream is = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+        while ((length = is.read(bytes)) >= 0) {
+            zipOutputStream.write(bytes, 0, length);
+        }
+        is.close();
+        byteArrayOutputStream.close();
```

Comment:
It feels a bit awkward to close the outputStream here rather than where it is opened.